lib/runtime/ppx_mysql_runtime.ml

```diff
@@ -1,3 +1,57 @@
+type column_error =
+  [ `Expected_non_null_column of int * string
+  | `Deserialization_error of int * string * string * string * string ]
+
+type 'a deserializer = string -> ('a, [`Deserialization_error of string]) result
+
+let wrap_failure of_string s =
+  match of_string s with
+  | v ->
+      Ok v
+  | exception Failure _ ->
+      Error (`Deserialization_error "cannot parse number")
+
+let string_of_string str = Ok str
+
+let int_of_string = wrap_failure Pervasives.int_of_string
+
+let int32_of_string = wrap_failure Int32.of_string
+
+let int64_of_string = wrap_failure Int64.of_string
+
+let bool_of_string str =
+  match Pervasives.int_of_string str with
+  | v ->
+      Ok (v <> 0)
+  | exception Failure _ ->
+      Error (`Deserialization_error "cannot parse boolean")
+
+let identity v = v
```

Comment:
Maybe use `%identity` here?

Reply:
I dunno, feels a bit too hackish and low-level. Moreover, the Brave New Flambda World will soon be upon us... ;-)

Reply:
The latest commit now uses `%identity`.